packages/devtools_app/lib/src/inspector/inspector_screen.dart

```diff
@@ -372,45 +350,95 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
   }
 }
 
-class ErrorNavigator extends StatelessWidget {
-  const ErrorNavigator(
-      {Key key,
-      @required this.selectedErrorIndex,
-      @required this.errorCount,
-      @required this.onSelectedErrorIndexChanged})
-      : super(key: key);
+class ErrorNavigator extends StatefulWidget {
+  const ErrorNavigator({
+    Key key,
+    @required this.errors,
+    @required this.selectedNode,
+    @required this.selectedErrorIndex,
+  }) : super(key: key);
+
+  final LinkedHashMap<String, InspectableWidgetError> errors;
+  final ValueListenable<InspectorTreeNode> selectedNode;
+  final ValueNotifier<int> selectedErrorIndex;
 
-  final int selectedErrorIndex;
-  final int errorCount;
-  final void Function(int) onSelectedErrorIndexChanged;
+  @override
+  _ErrorNavigatorState createState() => _ErrorNavigatorState();
+}
+
+class _ErrorNavigatorState extends State<ErrorNavigator> with AutoDisposeMixin {
+  @override
+  void initState() {
+    super.initState();
+    widget.selectedNode?.addListener(_selectedNodeChanged);
+  }
+
+  @override
+  void didUpdateWidget(covariant ErrorNavigator oldWidget) {
+    if (oldWidget.selectedNode != widget.selectedNode) {
+      oldWidget.selectedNode?.removeListener(_selectedNodeChanged);
+      widget.selectedNode?.addListener(_selectedNodeChanged);
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.selectedNode?.removeListener(_selectedNodeChanged);
+  }
+
+  void _selectedNodeChanged() {
+    final node = widget.selectedNode.value;
+    final inspectorRef = node?.diagnostic?.valueRef?.id;
+    // Check whether the node that was just selected has any errors associated
+    // with it.
+    var errorIndex = inspectorRef != null
+        ? widget.errors.keys.toList().indexOf(inspectorRef)
+        : null;
+    if (errorIndex == -1) {
+      errorIndex = null;
+    }
+
+    setState(() => widget.selectedErrorIndex.value = errorIndex);
+
+    // Additionally, mark this error as "read" and clear the badge.
+    if (errorIndex != null) {
+      serviceManager.errorBadgeManager.clearErrors(InspectorScreen.id);
```

Comment:
I don't think we should do this here. We should allow the screen to clear its errors once it is opened.

Reply:
This was to cover the case where you're already on the Inspector screen and you're reloading, and the badge increases. It felt expensive to be re-filtering the list on every error event, so I thought it may be nice to clear the badge if you then click on an error node (this felt like acknowledgement of the errors).

I wasn't entirely sure about it though, so happy to change (or remove)?